### ISSUE TITLE
a termination call to an orchestrator doesn't terminate any sub-orchestrations it started

### ISSUE THREAD

**Issue by peteroden** [Azure/azure-functions-durable-extension#153]
terminating an orchestrator should also terminate the sub-orchestrators it has started but this is not the case.

**Comment by kashimiz** [comment]
We might want to make this an optional behavior.

**Comment by peteroden** [comment]
That makes sense. I could see where in some scenarios you might want to leave a sub-orchestrator running. Perhaps handle it as an optional parameter on the terminate method?

**Comment by cgillum** [comment]
@simonporter what is the expected behavior in the Durable Task Framework? Should a terminate message terminate a parent orchestration and all it's children, or just the parent?

**Comment by peteroden** [comment]
@cgillum did you ever hear back from @simonporter on how this works in the Durable Task Framework?

**Comment by simonporter** [comment]
The current behavior in the default provider is that a terminate call on the parent will only terminate the parent, it does not propagate to the children.
It does look like a reasonable extension would be to add an option to also terminate sub-orchestrations. This might be an behavior not all providers can implement immediately.

**Comment by andreujuanc** [comment]
I'm starting to use DF in a more or less complex scenario, and this was driving me crazy!
Will have to redesign my architecture, since i was depending on other orchestrators. I will see if I can just have them as activities.

I think this is a very reasonable feature request, specially since the role of an orchestrator is to orchestrate!

**Comment by andreujuanc** [comment]
If we could have this https://github.com/Azure/azure-functions-durable-extension/issues/243
And if SubOrchestrations get passed it's parents instance id automatically, we could at least manually check for parent status and terminate accordingly.

**Comment by cgillum** [comment]
@andreujuanc can you open a separate issue to track making the parent instance ID available to the sub-orchestrator?  That might be a simple thing for us to add soon'ish.

**Comment by andreujuanc** [comment]
@cgillum  Done https://github.com/Azure/azure-functions-durable-extension/issues/246

**Comment by vany0114** [comment]
Hi @cgillum, @simonporter  just to confirm where did it land, the Orchestration Termination kills the sub-orchestrators as well? I can't find any documentation about that.

**Comment by yazansh** [comment]
Hello, I hope this issue gets resolved as soon as possible.
for me, I made a workaround that worked locally but unfortunately didn't work on the Azure portal.
In the same function app, I Created another HTTP triggered function, and using the durable orchestration client I used the get status async method "IList<DurableOrchestrationStatus> instances = await client.GetStatusAsync();", then iterated over each instance where orchestration runtime status is either "running" or "continued as new" and called "client.TerminateAsync(instanceId);".
